src/storage/src/hummock/shared_buffer/shared_buffer_manager.rs

```diff
@@ -347,32 +331,31 @@ impl SharedBufferManager {
 
     /// Deletes shared buffers before a given `epoch` exclusively.
     pub fn delete_before(&self, epoch: u64) {
-        let mut guard = self.shared_buffer.write();
-        let new = guard.split_off(&epoch);
-        *guard = new;
+        let mut inner = self.inner.write();
+        // `buffer` = newer part
+        let mut buffer = inner.shared_buffer.split_off(&epoch);
+        // `buffer` = older part
+        std::mem::swap(&mut inner.shared_buffer, &mut buffer);
+        // Update buffer size and scheduled size.
+        let deleted_size: usize = buffer
+            .into_iter()
+            .flat_map(|(_epoch, buffer)| buffer.into_iter().map(|(_end_key, batch)| batch.len()))
+            .sum();
+        self.scheduled_size
+            .fetch_sub(deleted_size, Ordering::Release);
```

Comment:
IIUC, `scheduled_size` represent buffer size in `SharedBufferUploader` queue. Should we put it in `SharedBufferUploader` and subtract `scheduled_size` after the corresponding buffer is uploaded instead of after it is deleted?

Reply:
`scheduled_size` contains uploading and uploaded but not cleared buffer size rather than uploading buffer size. It will only be subtracted when cleared.